src/libraries/Microsoft.Bcl.Cryptography/src/System/Security/Cryptography/X509Certificates/X509CertificateKeyAccessors.cs

```diff
@@ -15,6 +17,100 @@ namespace System.Security.Cryptography.X509Certificates
     /// </summary>
     public static class X509CertificateKeyAccessors
     {
+        /// <summary>
+        ///   Gets the <see cref="MLKem"/> public key from this certificate.
+        /// </summary>
+        /// <param name="certificate">
+        ///   The X.509 certificate that contains the public key.
+        /// </param>
+        /// <returns>
+        ///   The public key, or <see langword="null"/> if this certificate does not have an ML-KEM public key.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="certificate"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The certificate has an ML-KEM public key, but the platform does not support ML-KEM.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   The public key was invalid, or otherwise could not be imported.
+        /// </exception>
+        [ExperimentalAttribute(Experimentals.PostQuantumCryptographyDiagId, UrlFormat = Experimentals.SharedUrlFormat)]
+        public static MLKem? GetMLKemPublicKey(this X509Certificate2 certificate)
+        {
+#if NET10_0_OR_GREATER
+            return certificate.GetMLKemPublicKey();
+#else
```

Comment:
This basically won't get exercised by tests for now because we have no platforms where `MLKem.IsSupported` is `true` for M.B.C that won't type-forward. If we are more comfortable taking this as PNSE until we get .NET Framework lit up, we can do that, too.